projects/plugins/boost/app/assets/src/js/features/speed-score/pop-out/pop-out.tsx

```diff
@@ -0,0 +1,152 @@
+	id: 'score-increase',
+	title: __( 'Your site got faster', 'jetpack-boost' ),
+	body: __( `That's great! If you’re happy, why not rate Boost?`, 'jetpack-boost' ),
+	cta: __( 'Rate the Plugin', 'jetpack-boost' ),
+	ctaLink: 'https://wordpress.org/support/plugin/jetpack-boost/reviews/#new-post',
+};
+
+const slowerMessage = {
+	id: 'score-decrease',
+	title: __( 'Speed score has fallen', 'jetpack-boost' ),
+	body: (
+		<>
+			<p>
+				{ __(
+					'Most of the time Jetpack Boost will increase your site speed, but there may be cases where your score does not increase.',
+					'jetpack-boost'
+				) }
+			</p>
+			<p>
+				{ __(
+					'Try refreshing your score, and if it doesn’t help, check our guide on improving your site speed score:',
+					'jetpack-boost'
+				) }
+			</p>
+		</>
+	),
+	cta: __( 'Read the guide', 'jetpack-boost' ),
+	ctaLink: 'https://jetpack.com/support/speed-up-your-site/',
+};
+
+/**
+ * Wrapper for PopOut which provides a data sync context. Can be removed once we know the
+ * parent of PopOut is always wrapped in a DataSyncProvider.
+ *
+ * @param {Props} props Properties.
+ */
+export default function PopOut( props: Props ) {
+	return (
+		<DataSyncProvider>
+			<_PopOut { ...props } />
+		</DataSyncProvider>
+	);
+}
+
+/**
+ * Helper hook - Use data sync to track which score popouts have been dismissed.
+ *
+ * @param id - The id of the score popout to track. Accepts null (which will result in dummy return values) for ease of use.
+ */
+function useDismissedScoreAlerts( id: string | null ): [ boolean, () => void ] {
+	const [ { data }, { mutate } ] = useDataSync(
+		'jetpack_boost_ds',
+		'dismissed_score_prompt',
+		z.array( z.string() )
+	);
+
+	// If no id provided, give back a dummy function with "yes, dismissed".
+	if ( ! id ) {
+		return [ true, () => {} ];
+	}
+
+	const isDismissed = data?.includes( id ) || false;
+	const dismiss = () => {
+		if ( ! isDismissed ) {
+			mutate( [ ...( data || [] ), id ] );
+		}
+	};
+
+	return [ isDismissed, dismiss ];
+}
+
+function _PopOut( { scoreChange, onClose }: Props ) {
+	const [ message, setMessage ] = useState< ScoreChangeMessage | null >( null );
+	const hasScoreChanged = scoreChange !== false && Math.abs( scoreChange ) > 5;
+
+	// Keep the last score change message in state, so we can animate it out after the user dismisses it.
+	useEffect( () => {
+		if ( hasScoreChanged ) {
+			setMessage( scoreChange > 0 ? fasterMessage : slowerMessage );
+		}
+	}, [ hasScoreChanged, scoreChange ] );
+
+	// Use datasync to track which score alerts have been dismissed.
+	const [ isDismissed, dismiss ] = useDismissedScoreAlerts( message?.id );
+
+	// Callback to handle the user dismissing the message by clicking the CTA or the dismiss option
+	function handleDismiss() {
+		dismiss();
+		onClose();
+	}
+
+	return (
+		<div id="parent" className={ styles.wrapper }>
+			<CSSTransition
+				in={ hasScoreChanged && ! isDismissed }
+				appear={ true }
+				classNames={ { ...styles } }
```

Comment:
Does that mean all the CSS classes will be applied to this component, even headline, card, etc?

Reply:
No, they won't all be used. `CSSTransition` expects an object where keys include things like this to define css classes for different parts of the transition:
 
```
classNames={{
 appear: 'my-appear',
 appearActive: 'my-active-appear',
 appearDone: 'my-done-appear',
 enter: 'my-enter',
 enterActive: 'my-active-enter',
 enterDone: 'my-done-enter',
 exit: 'my-exit',
 exitActive: 'my-active-exit',
 exitDone: 'my-done-exit',
}}
```

[Their docs ](https://reactcommunity.org/react-transition-group/css-transition) recommend just sending the whole style object when using a css module, in which case only styles in the module scss file named things like `.appearDone` and others from the above list will be used.